src/test/java/com/greenfox/lvlup/controller/BadgeControllerTest.java

```diff
@@ -0,0 +1,50 @@
+package com.greenfox.lvlup.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(BadgeController.class)
+public class BadgeControllerTest {
+
+  @Autowired
+  public MockMvc mvc;
+
+  private UserBadgeSetDTO testBadgeSet = new UserBadgeSetDTO();
+  private String tokenString = "testToken";
+
+  @Test
+  public void showBadgesTestWithCorrectHeader() throws Exception {
+    mvc.perform(get("/badges")
+        .header("userTokenAuth", tokenString)
+        .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(content().string(stringify(testBadgeSet)));
+  }
+
+  @Test
+  public void showBadgesTestWithoutToken() throws Exception {
+    mvc.perform(get("/badges")
+        .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isUnauthorized())
+        .andExpect(jsonPath("$.error").value("Unauthorized"));
+  }
+
+  private String stringify(Object object) throws JsonProcessingException {
```

Comment:
please move this function to a generic place, it will be used in different files. you can create a static utils class or a helper, up to you. just find a proper place to it